airflow/providers/google/cloud/transfers/sql_to_gcs.py

```diff
@@ -232,6 +232,10 @@ def _write_local_data_files(self, cursor):
 
         return files_to_upload
 
+    def get_cursor_iterator(self, cursor):
```

Comment:
Why is it necessary? [pep-0249](https://www.python.org/dev/peps/pep-0249/) recommends that cursor have a `__iiter__` method, so this should work without it. If the Vertica library does not implement it, I think it is worth fixing the problem and not try to change the interface of other classes. the adapter design pattern can be helpful here. See: 
https://github.com/apache/airflow/blob/2bac4810a48d43692be8f585167925e2c48abb46/airflow/providers/google/cloud/transfers/presto_to_gcs.py#L27

Reply:
As per pep-0249, `__iter__` is an optional dbapi extension. So, other datastores also may or may not choose to implement this.

The above change will not change the existing functionality, rather will give the flexibility to the user to define/modify the iterator if in case it is not provided by the datastore cursor.

I am fine with making changes. Kindly let me know.

Reply:
I think that PEP documents should also be treated as a recommendation and good practice. In this case, PEP recommends creating an interface with  `__iter__` method instead of alternative solutions such as the get_iter methods.

Reply:
If it is possible, we should stick to the PEP recommendations. When this is not possible, we should use our own solutions, but I think that here it is possible to achieve compatibility with PEP.

Reply:
I was looking at `_PrestoToGCSPrestoCursorAdapter` code, it looks like the whole cursor code is copied from [presto code](https://github.com/prestodb/presto-python-client/blob/master/prestodb/dbapi.py#L170).

I am afraid if we do the same, then in the future, for some reason, logic changes, then we will also have to change the logic at our end also. Isn't that overhead?